tests_e2e/pipeline/scripts/execute_tests.sh

```diff
@@ -2,36 +2,62 @@
 
 set -euxo pipefail
 
+#
+# Set the correct mode for the private SSH key and generate the public key.
+#
+cd "$HOME"
+mkdir ssh
+cp "$DOWNLOADSSHKEY_SECUREFILEPATH" ssh
+chmod 700 ssh/id_rsa
+ssh-keygen -y -f ssh/id_rsa > ssh/id_rsa.pub
+
+#
+# Change the ownership of the "ssh" directory we just created, as well as the sources and staging directories.
+# Make waagent (UID 1000 in the container) the owner of both locations, so that it can write to them.
+# This is needed because building the agent package writes the egg info to the source code directory, and
+# tests write their logs to the staging directory.
+#
+sudo find ssh -exec chown 1000 {} \;
+sudo chown 1000 "$BUILD_SOURCESDIRECTORY"
+sudo chown 1000 "$BUILD_ARTIFACTSTAGINGDIRECTORY"
+
+#
 # Pull the container image used to execute the tests
+#
 az login --service-principal --username "$AZURE_CLIENT_ID" --password "$AZURE_CLIENT_SECRET" --tenant "$AZURE_TENANT_ID" > /dev/null
 
 az acr login --name waagenttests
 
 docker pull waagenttests.azurecr.io/waagenttests:latest
 
-# Building the agent package writes the egg info to the source code directory, and test write their logs to the staging directory.
-# Make waagent (UID 1000 in the container) the owner of both locations, so that it can write to them.
-sudo chown 1000 "$BUILD_SOURCESDIRECTORY"
-sudo chown 1000 "$BUILD_ARTIFACTSTAGINGDIRECTORY"
-
 # A test failure will cause automation to exit with an error code and we don't want this script to stop so we force the command
 # to succeed and capture the exit code to return it at the end of the script.
 echo "exit 0" > /tmp/exit.sh
 
 docker run --rm \
-      --volume "$BUILD_SOURCESDIRECTORY:/home/waagent/WALinuxAgent" \
-      --volume "$DOWNLOADSSHKEY_SECUREFILEPATH:/home/waagent/id_rsa" \
-      --volume "$BUILD_ARTIFACTSTAGINGDIRECTORY:/home/waagent/logs" \
-      --env SUBSCRIPTION_ID \
-      --env AZURE_CLIENT_ID \
-      --env AZURE_CLIENT_SECRET \
-      --env AZURE_TENANT_ID \
-      waagenttests.azurecr.io/waagenttests \
-      bash --login -c \
-        "\$HOME/WALinuxAgent/tests_e2e/orchestrator/scripts/run-scenarios -t $TEST_SUITES -l $COLLECT_LOGS -k $SKIP_SETUP" \
```

Comment:
The TEST_SUITES parameter needs to be surrounded by quotes, in order to allow spaces in its value.

Previously we were invoking the run-scenarios scripts, which was basically a pass thru to LISA. With the need to pass the pipeline parameters to the LISA runbook, that script was getting over-complicated. It is just simpler to call LISA here. I did that, and removed run-scenarios.